pkg/ui/spinner_test.go

```diff
@@ -0,0 +1,97 @@
+package ui_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	"groundcover.com/pkg/ui"
+)
+
+type SpinnerTestSuite struct {
+	suite.Suite
+}
+
+func (suite *SpinnerTestSuite) SetupSuite() {}
+
+func (suite *SpinnerTestSuite) TearDownSuite() {}
+
+func TestSpinnerSuite(t *testing.T) {
+	suite.Run(t, &SpinnerTestSuite{})
+}
+
+func (suite *SpinnerTestSuite) TestPollFuncSuccues() {
```

Comment:
see if you can use `https://github.com/benbjohnson/clock` lib instead of really waiting

Reply:
https://github.com/benbjohnson/clock#controlling-time
you need to use mock.Ticker which will not test the function